### PR TITLE
[CST] Upgrade to react-router v6

### PR DESCRIPTION
**Pull request by jerekshoe**
## Summary
Upgrading the version of react-router used to v6. High level overview of the changes:
* `Route` components must be a child of a single `Routes` component parent
* There is no longer an `IndexRoute` component. It is recommended to create a route that has an `element` prop set to a `Navigate` component that redirects the user to the appropriate route
* Instead of using `this.props.children` to render child routes, components must use the `Outlet` component
* Passing props to child routes is possible via the `Outlet` components `context` prop
* Retrieving props passed down from a parent route is possible via the `useOutletContext` hook
* Links now use relative routes by default. Absolute routing is done by prepending a `/` to relative routes
* Added a `withRouter` HOC that is responsible for passing basic router hooks `params`, `navigate`, `location` to it's children. This is useful for using hooks with Class-based components
* Added a NavLink component that replaces the IndexLink component as it has been removed from `react-router`
* Instances of `this.props.router.push(newUrl)` have been replaced with `navigate(newNewUrl)`
* Instances of `this.props.router.replace(newUrl)` have been replaced with `navigate(newNewUrl, { replace: true })`;

#### ClaimCardLink
1) Navigate to the claim list page (`/track-claims/your-claims`)
2) Each card should have a link labeled 'View Details'
3) Clicking on one of the links should navigate you to the details page for a claim
4) You need to test the 3 different types of claim cards:
  * Claims -> Card title should start with 'Claim for'
  * STEM Claims -> Card title should start with 'EDITH'
  * Appeals -> Card title should start with 'Appeal for'

#### FilesNeeded
**Note:** You will need to find a claim that has an open first-party evidence request to test this one
1) Navigate to the files tab for a claim that has an open first-party evidence request
2) You should see a yellow warning alert with a `View Details` call to action link
   * Take note of the text content of the heading of the alert
3) Clicking the `View Details` button should take you to a page that has a heading that matches the heading from the previous alert (URL should be something like `/track-claims/your-claims/009_ClaimWithTrackedItems/document-request/397176`)

#### FilesOptional
**Note:** You will need to find a claim that has an open third-party evidence request to test this one
1) Navigate to the files tab for a claim that has an open third-party evidence request
2) You should see a blue info alert with a `add it here` link
   * Take note of the text content of the heading of the alert
3) Clicking the `View Details` button should take you to a page that has a heading that matches the heading from the pervious alert (URL should be something like `/track-claims/your-claims/009_ClaimWithTrackedItems/document-request/397176`)

#### ClosedClaimAlert
1) Navigate to the details for a closed claim that has the `decision letter ready` indicator on the claim card
2) You should see a call to action link in the blue info alert.
3) Clicking that should take you to the page titled 'Your VA claim letters'

#### RecentActivity
**Note:** You will need to find a claim that has an open third-party evidence request to test this one
1) Navigate to the status tab for a claim that has an open third-party evidence request
2) Scroll down to the 'Recent Activity' section. You should see an entry that has a slim blue info alert. This alert should have a link labeled `add it here`
3) Clicking the link should navigate you to the document request page. (URL should be something like `/track-claims/your-claims/009_ClaimWithTrackedItems/document-request/397176`)

#### What we are doing
1) Navigate to the status tab for an open claim
2) You should see a section titled 'What we're doing'
3) Under the heading there should be a card that has a link labeled 'Overview of the process'
4) Clicking the link should take you to the 'Overview' tab (URL should be something like: `/track-claims/your-claims/009_ClaimWithTrackedItems/overview`)

#### AddFilesFormOld
1) Navigate to the files tab for an open claim
2) Scroll down to the `Additional Evidence` section
3) You should see a link labeled `Cancel` besides the `Submit Files for Review` button
4) Clicking that should redirect you back to the tab you were on previously
  * If you navigated to the files tab via URL, clicking cancel should have no effect

#### AskVAToDecide
1) Navigate to the details for a claim that is in the 'Evidence Gathering, Review, Decision' step
2) Navigate to the 'Files' tab
3) If you scroll down you should see a blue box with the heading `Ask for your claim decision`
4) Click the `View Details` button
5) This should navigate you to a page that has the heading "Ask for your claim decision"

#### ClaimsDecision
1) Navigate to the details for a closed claim that has the `decision letter ready` indicator
2) You should see a call to action link in the blue info alert.
3) Clicking that should take you to the page titled 'Your VA claim letters'

#### ClosedClaimMessage
**Note:** This is much easier to test locally then it is in staging
1) When on the claims list page (`/track-claims/your-claims`), you should see a yellow warning alert directly under the `Your claims, decision reviews, or appeals' heading. There should be a link with the text 'Your compensation Received June 2, 2018'
2) Clicking that link should take you to the status tab for that claim (something like `/track-claims/your-claims/006_ClosedClaim/status`)

#### FilesNeededOld
**Note:** You will need to find a claim that has an open first-party evidence request to test this one
1) Navigate to the files tab for a claim that has an open first-party evidence request
2) You should see a yellow rectangle with a `View Details` button
   * Take note of the text content of the heading of the alert
3) Clicking the `View Details` button should take you to a page that has a heading that matches the heading from the pervious alert (URL should be something like `/track-claims/your-claims/009_ClaimWithTrackedItems/document-request/397176`)

#### FilesOptionalOld
**Note:** You will need to find a claim that has an open third-party evidence request to test this one
1) Navigate to the files tab for a claim that has an open third-party evidence request
2) You should see a grey rectangle with a `View Details` button
   * Take note of the text content of the heading of the alert
3) Clicking the `View Details` button should take you to a page that has a heading that matches the heading from the pervious alert (URL should be something like `/track-claims/your-claims/009_ClaimWithTrackedItems/document-request/397176`)

#### NeedFilesFromYou
**Note:** You will need to find a claim that has an open first-party evidence request to test this one
1) Navigate to the status tab for a claim that has an open first-party evidence request
2) You should see a warning alert with a `View Details` button
   * Take note of the text content of the heading of the alert
3) Clicking the `View Details` button should take you to the 'Files' tab

## Related issue(s)

- _Link to ticket created in va.gov-team repo_
department-of-veterans-affairs/va.gov-team#50486

## Testing done
Updated tests to work with `react-router-dom-v5-compat`. Many of the changes were around the fact that when using hooks like `useNavigate` and `useLocation`, there must be a Router component present. This also applies to any component that use the `Link` component as well.

## What areas of the site does it impact?
Claim Status Tool. This is a major update that touches almost every part of the application

### Quality Assurance & Testing

- [ ] I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [ ] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [ ] Linting warnings have been addressed
- [ ] Documentation has been updated ([link to documentation](#) \*if necessary)
- [ ] Screenshot of the developed feature is added
- [ ] [Accessibility testing](https://depo-platform-documentation.scrollhelp.site/developer-docs/wcag-2-1-success-criteria-and-foundational-testing) has been performed

### Error Handling

- [ ] Browser console contains no warnings or errors.
- [ ] Events are being sent to the appropriate logging solution
- [ ] Feature/bug has a monitor built into Datadog or Grafana (if applicable)

### Authentication

- [ ] Did you login to a local build and verify all authenticated routes work as expected with a test user

### :warning: Team Sites (only applies to modifications made to the VA.gov header) :warning:

- [ ] The vets-website header does not contain any web-components
- [ ] I used the [proxy-rewrite steps](https://github.com/department-of-veterans-affairs/vets-website/tree/main/src/applications/proxy-rewrite#local-dev) to test the injected header scenario
- [ ] I reached out in the `#sitewide-public-websites` Slack channel for questions

## Requested Feedback

(OPTIONAL) _What should the reviewers know in addition to the above. Is there anything specific you wish the reviewer to assist with. Do you have any concerns with this PR, why?_

### CODE REVIEW

File: src/applications/claims-status/tests/components/claim-status-tab/WhatWeAreDoing.unit.spec.jsx [hunk 1]
```diff
@@ -1,40 +1,38 @@
...
-          latestPhaseType: 'INITIAL_REVIEW',
-          previousPhases: {
-            phase1CompleteDate: '2023-02-08',
-            phase2CompleteDate: '2023-02-08',
-          },
-        },
-      },
-    };
-    const { container, getByText } = render(<WhatWeAreDoing claim={claim} />);
-    expect($('.what-were-doing-container', container)).to.exist;
```
Comment (jerekshoe) [inline src/applications/claims-status/tests/components/claim-status-tab/WhatWeAreDoing.unit.spec.jsx:36]: Removed this as the other 2 assertions will not pass if this one fails

File: src/applications/claims-status/tests/components/claim-files-tab/FilesOptional.unit.spec.jsx [hunk 2]
```diff
@@ -1,28 +1,22 @@
...
-    );
+    const { getByText } = renderWithRouter(<FilesOptional item={item} />);
 
     getByText(item.displayName);
     getByText(item.description);
     getByText(
       'You don’t have to do anything, but if you have this information you can',
     );
     getByText('add it here.');
-    expect($('va-alert', container)).to.exist;
```
Comment (jerekshoe) [inline src/applications/claims-status/tests/components/claim-files-tab/FilesOptional.unit.spec.jsx:26]: The other assertions will not pass if this one doesn't

File: src/applications/claims-status/tests/components/ClaimsStatusApp.unit.spec.jsx [hunk 3]
```diff
@@ -49,9 +19,13 @@ describe('<ClaimsStatusApp>', () => {
 
   it('should render children if feature toggles are available', () => {
```
Comment (pmclaren19) [inline src/applications/claims-status/tests/components/ClaimsStatusApp.unit.spec.jsx:20]: rename test to not have children

File: src/applications/claims-status/tests/components/FilesOptionalOld.unit.spec.jsx [hunk 4]
```diff
@@ -10,7 +10,7 @@ describe('<FilesOptionalOld>', () => {
       displayName: 'Request 1',
       description: 'This is a alert',
     };
-    const screen = render(<FilesOptionalOld id={id} item={item} />);
```
Comment (pmclaren19) [inline src/applications/claims-status/tests/components/FilesOptionalOld.unit.spec.jsx:13]: remove the id and you can move the item outside the file

File: src/applications/claims-status/tests/components/appeals-v2/AppealsV2TabNav.unit.spec.jsx [hunk 5]
```diff
@@ -1,31 +1,22 @@
...
   it('should render', () => {
-    const wrapper = shallow(<AppealsV2TabNav {...defaultProps} />);
-    expect(wrapper.type()).to.equal('nav');
-    wrapper.unmount();
+    const { container } = renderWithRouter(<AppealsV2TabNav />);
+
+    expect($('ul.tabs', container)).to.exist;
   });
 
   it('should render 2 tabs: Status and Detail', () => {
```
Comment (pmclaren19) [inline src/applications/claims-status/tests/components/appeals-v2/AppealsV2TabNav.unit.spec.jsx:16]: rename this should say Issues

**Review by pmclaren19 — APPROVED** [review]
Paired with Jerek. Went to his branch and verified all links in CST are working as expected. 
Afterwards went through his code and verified all looked good! 

I think this is good to merge 👍

**Pull request merged**